### PR TITLE
[WIP] Replace var with const/let, add eslint recommended rules. Fixes #899

### LINKED ISSUE #899: Convert var to const/let
var has some weird behaviors (hoisting to top of the function, function-scoped rather than block-scoped) that `const` and `let` avoid. Eslint has a [rule](https://eslint.org/docs/rules/no-var) to enable this, and can auto-fix it.

### PR DESCRIPTION
**Pull request by turnerhayes**

### CODE REVIEW

**Comment by knod** [comment]
Hmm, didn't realize it was introducing a lot of new rules. Making this a WIP until we have time to take it in.

**Comment by knod** [comment]
WIP's should not be merged.

**Comment by knod** [comment]
Sorry that I didn't notice the other rule changes before.

File: .eslintrc.json [hunk 1]
```diff
@@ -48,6 +51,10 @@
     "react/jsx-tag-spacing": ["warn"],
     "react/self-closing-comp": "warn",
     "space-in-parens": ["warn", "never"],
-    "space-infix-ops": ["warn"]
+    "space-infix-ops": ["warn"],
+    "no-var": ["error"],
+    "no-undef": ["error"],
+    "no-unused-vars": ["error"],
```
Comment (knod) [inline .eslintrc.json:57]: Can the linter automatically fix `no-unused-vars`? Same question for the other config options here. Our decision currently is that linting issues shouldn't block a commit or PR.

Comment (turnerhayes) [reply .eslintrc.json:57]: It looks like it can fix some (most)? `no-var`s, but not `no-undef` or `no-unused-vars`.

**Pull request closed without merging**